uPortal-layout/uPortal-layout-core/src/main/java/org/apereo/portal/layout/dao/jpa/StylesheetDescriptorImpl.java

```diff
@@ -304,7 +304,7 @@ public ILayoutAttributeDescriptor getLayoutAttributeDescriptor(String name) {
         }
 
         //Remove all old data entries that were not updated
-        dataMap.keySet().remove(oldDataKeys);
+        dataMap.keySet().removeAll(oldDataKeys);
```

Comment:
`dataMap.keySet()` and `oldDataKeys` are both`Set<String>`.
Remove would try find a `Set<String>` in a `String` which will never be found.
`removeAll` removes the sub`Set` from the `Set`.